python/setup.py

```diff
@@ -354,6 +468,19 @@ def append_cmake_bool(value, varname):
                 shutil.move(pjoin(build_prefix, 'include'),
                             pjoin(build_lib, 'pyarrow'))
 
+                # We need to, again, add the PyArrow cpp include folder
```

Comment:
It is because of these lines right before:

https://github.com/apache/arrow/blob/66cd4f9dffb067deed9af0bdb805e99d68acbcc0/python/setup.py#L464-L469

where the whole include folder is deleted in case of re-build. As the pyarrow cpp includes are added in the previous step (`_run_cmake_pyarrow_cpp`) they always get deleted. But are also needed when the `_run_cmake` starts.

Will add some more info to the comment +1